File: 001-DataAccessLayer/Models/MyData.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace UpstartTest
{
	public class MyData
	{
		private Guid _id;
		private string _name;
		private string _description;
		private int _amount;
		private long _date;
		private bool _isPrivate;

		public Guid id
		{
			get { return _id; }
			set { _id = value; }
		}

		[Required(ErrorMessage = "Missing name.")]
		[StringLength(100, ErrorMessage = "Name can't exceeds 100 chars.")]
		public string name
		{
			get { return _name; }
			set { _name = value; }
		}

		public string description
		{
			get { return _description; }
			set { _description = value; }
		}

		[Range(1, 999999)]
		public int amount
		{
			get { return _amount; }
			set { _amount = value; }
		}

		public long date
		{
			get { return _date; }
			set { _date = value; }
		}

		public bool isPrivate
		{
			get { return _isPrivate; }
			set { _isPrivate = value; }
		}


		public override string ToString()
		{
			return
				id + " " +
				name + " " +
				description + " " +
				date + " " +
				isPrivate;
		}
	}
}

File: UpstartTestAngular/src/app/models/MyData.ts
export class MyData {
    // private _id: string;
    // private _name: string;
    // private _description: string;
    // private _amount: number;
    // private _date: Date;
    // private _isPrivate: boolean;

    public constructor(
        public id?: string,
        public name?: string,
        public description?: string,
        public amount?: number,
        public date?: Date,
        public isPrivate?: boolean
    ) { 
        // this.id = id;
        // this.name = name;
        // this.description = description;
        // this.amount = amount;
        // this.date = date;
        // this.isPrivate = isPrivate;
    }

    // get id():string{
    //     console.debug("get id: " + JSON.stringify(this._id));
    //     return this._id;
    // }

    // set id(val){
    //     this._id=val;
    //     console.debug("set id: " + JSON.stringify(this._id));
    // }

    // get name():string{
    //     console.debug("get name: " + JSON.stringify(this._name));
    //     return this._name;
    // }

    // set name(val){
    //     this._name=val;
    //     console.debug("set name: " + JSON.stringify(this._name));
    // }

    // get description():string{
    //     console.debug("get description: " + JSON.stringify(this._description));
    //     return this._description;
    // }

    // set description(val){
    //     this._description=val;
    //     console.debug("set description: " + JSON.stringify(this._description));
    // }

    // get amount():number{
    //     console.debug("get amount: " + JSON.stringify(this._amount));
    //     return this._amount;
    // }

    // set amount(val){
    //     this._amount=val;
    //     console.debug("set amount: " + JSON.stringify(this._amount));
    // }

    // get date():Date{
    //     console.debug("get date: " + JSON.stringify(this._date));
    //     if(this._date instanceof Date){
    //         return this._date;
    //     }
	// 	return new Date();
    // }

    // set date(val){
    //     this._date=new Date(val);
    //     console.debug("set date: " + JSON.stringify(this._date));
    // }

    // get isPrivate():boolean{
    //     console.debug("get isPrivate: " + JSON.stringify(this._isPrivate));
    //     return this._isPrivate;
    // }

    // set isPrivate(val){
    //     this._isPrivate=val;
    //     console.debug("set isPrivate: " + JSON.stringify(this._isPrivate));
    // }
}